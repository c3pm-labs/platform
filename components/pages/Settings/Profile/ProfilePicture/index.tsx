import React from 'react';
import {
  Button,
  makeStyles, Typography,
} from '@material-ui/core';

import Avatar from 'components/Avatar';

import SettingWrapper from '../../SettingWrapper';
import UpdateButton from '../../UpdateButton';

const useStyles = makeStyles((theme) => ({
  stylePicture: {
    width: '5.5em',
    height: '5.5em',
  },
  containerProfilePicture: {
    display: 'flex',
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    height: '100%',
    maxHeight: '90px',
    [theme.breakpoints.down('xs')]: {
      flex: 1,
      width: '100%',
      maxHeight: '300px',
      marginBottom: theme.spacing(2),
    },
  },
  containerPadding: {
    display: 'flex',
    flex: 1,
    paddingLeft: theme.spacing(5),
    flexDirection: 'row',
    justifyContent: 'space-between',
    [theme.breakpoints.down('xs')]: {
      paddingLeft: 0,
      justifyContent: 'space-around',
      flexDirection: 'column',
    },
    alignItems: 'center',
    height: '100%',
  },
  containerUplaod: {
    display: 'flex',
    flex: 0.9,
    flexDirection: 'column',
    justifyContent: 'space-around',
    alignContent: 'flex-start',
    height: '90%',
    paddingRight: theme.spacing(1),
    paddingLeft: theme.spacing(1),
    [theme.breakpoints.down('xs')]: {
      paddingRight: 0,
      flex: 0.7,
    },
  },
  textButton: {
    color: 'white',
    fontWeight: 500,
    textTransform: 'capitalize',
  },
  customButton: {
    display: 'flex',
    maxHeight: '36px',
    maxWidth: '180px',
    [theme.breakpoints.down('xs')]: {
      maxWidth: '0px',
    },
    minWidth: '120px',
  },
  customPicture: {
    display: 'flex',
    flex: 0.45,
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  description: {
    paddingTop: theme.spacing(2),
    display: 'flex',
    fontWeight: 500,
    opacity: 0.6,
    fontSize: '1.2em',
    color: `${theme.palette.text.primary}`,
  },
}));

function ProfilePicture(): JSX.Element {
  const classes = useStyles();
  const viewer = {
    id: 'string',
    username: 'Alice',
    email: 'alice.dupont@gmail.com',
    description: 'Software developper',
    packages: [],
  };
  const onClick = (): void => (console.log('upload'));

  return (
    <SettingWrapper
      title="Profile Picture :"
      // saveDescription="Save your new picture"
      // onSave={(): void => (console.log('save'))}
    >
      <div className={classes.containerProfilePicture}>
        <div className={classes.containerPadding}>
          <div className={classes.customPicture}>
            <Avatar
              stylePicture={classes.stylePicture}
              user={viewer}
              withName={false}
            />
            <div className={classes.containerUplaod}>
              <Button
                component={Button}
                title="Button"
                variant="contained"
                color="primary"
                fullWidth
                className={classes.customButton}
                onClick={onClick}
              >
                <Typography className={classes.textButton}>
                  Upload Image
                </Typography>
              </Button>
              <Typography className={classes.description}>
                Customize your profile picture
              </Typography>
            </div>
          </div>
          <UpdateButton description="Update your profil" onClick={(): void => (console.log('save'))} />
        </div>
      </div>
    </SettingWrapper>
  );
}

export default ProfilePicture;
