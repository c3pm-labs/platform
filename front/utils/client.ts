import { ApolloClient, InMemoryCache, createHttpLink } from '@apollo/client'
import { setContext } from '@apollo/client/link/context'

const httpLink = createHttpLink({
  uri: process.env.GRAPHQL_URL,
  credentials: 'include',
})

const authLink = setContext((_, { headers }) => {
  return {
    headers: {
      ...headers,
    },
  }
})

const client = new ApolloClient({
  link: authLink.concat(httpLink),
  cache: new InMemoryCache(),
  ssrMode: typeof window === 'undefined',
})

export default client
