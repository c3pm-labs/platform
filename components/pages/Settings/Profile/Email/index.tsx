import React from 'react';
import {
  makeStyles,
} from '@material-ui/core';
import { Formik, Form } from 'formik';
import * as yup from 'yup';
import { UPDATE } from 'queries';
import { useMutation } from '@apollo/client';
// import { useRouter } from 'next/router';
// import { useUser } from 'hooks/user';

import InputFied from '../../InputField';
import SettingWrapper from '../../SettingWrapper';
import UpdateButton from '../../UpdateButton';
import { initialValues, UpdateUserParams } from '../utils/UpdateUtils';

const useStyles = makeStyles((theme) => ({
  containerInput: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'space-between',
    alignItems: 'flex-start',
    [theme.breakpoints.down('xs')]: {
      flex: 1,
      width: '100%',
      marginBottom: theme.spacing(3),
      marginTop: theme.spacing(3),
    },
  },
  styleForm: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    flex: 1,
    width: '100%',
    [theme.breakpoints.down('xs')]: {
      flexDirection: 'column',
    },
  },
}));

const testSchema = yup.object().shape({
  email: yup.string().email('Email is invalid').required('Email is required'),
});

function Email(): JSX.Element {
  const classes = useStyles();
  // const router = useRouter();
  // const user = useUser({ id: String(router.query.params) });
  const [update] = useMutation(UPDATE);

  return (
    <SettingWrapper
      title="Email :"
    >
      <Formik
        validationSchema={testSchema}
        initialValues={initialValues}
        onSubmit={(values: UpdateUserParams): void => {
          update({
            variables: {
              id: 'ckh6whm390001eyt8mg7ja97m', // CHANGER PAR LE VRAI ID, PROBLEME DE LOGIN!!!
              email: values.email,
            },
          });
        }}
      >
        <div className={classes.containerInput}>
          <Form className={classes.styleForm}>
            <InputFied name="email" label="Email :" value="alice.dupont@gmail.com" />
            <UpdateButton description="Update your email" onClick={(): void => (console.log('save'))} />
          </Form>
        </div>
      </Formik>
    </SettingWrapper>
  );
}

export default Email;
