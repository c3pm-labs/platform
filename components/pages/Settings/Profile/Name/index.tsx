import React from 'react';
import {
  makeStyles,
} from '@material-ui/core';
import { Formik, Form } from 'formik';
import * as yup from 'yup';
import { UPDATE } from 'queries';
import { useMutation } from '@apollo/client';
// import { useRouter } from 'next/router';
// import { useUser } from 'hooks/user';

import InputFied from '../../InputField';
import SettingWrapper from '../../SettingWrapper';
import UpdateButton from '../../UpdateButton';
import { initialValues, UpdateUserParams } from '../utils/UpdateUtils';

const useStyles = makeStyles((theme) => ({
  containerInput: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'space-between',
    alignItems: 'flex-start',
    minHeight: '100px',
    height: '100px',
    maxHeight: '100px',
    [theme.breakpoints.down('xs')]: {
      flex: 1,
      width: '100%',
      maxHeight: '300px',
    },
  },
  styleForm: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    flex: 1,
    width: '100%',
    [theme.breakpoints.down('xs')]: {
      flexDirection: 'column',
    },
  },
  textButton: {
    color: 'white',
    fontWeight: 500,
    textTransform: 'capitalize',
  },
  customButton: {
    maxHeight: '36px',
  },
}));

const testSchema = yup.object().shape({
});

function Name(): JSX.Element {
  const classes = useStyles();
  // const router = useRouter();
  // const user = useUser({ id: String(router.query.params) });
  const [update] = useMutation(UPDATE);
  return (
    <SettingWrapper
      title="Name :"
    >
      <Formik
        validationSchema={testSchema}
        initialValues={initialValues}
        onSubmit={(values: UpdateUserParams): void => {
          update({
            variables: {
              id: 'ckh6whm390001eyt8mg7ja97m', // CHANGER PAR LE VRAI ID, PROBLEME DE LOGIN!!!
              newUsername: values.newUsername,
            },
          });
        }}
      >
        <div className={classes.containerInput}>
          <Form className={classes.styleForm}>
            <InputFied
              name="newUsername"
              label="Username :"
              value="Alice"
              multiline
            />
            <UpdateButton description="Update your name" onClick={(): void => (console.log('save'))} />
          </Form>
        </div>
      </Formik>
    </SettingWrapper>
  );
}

export default Name;
