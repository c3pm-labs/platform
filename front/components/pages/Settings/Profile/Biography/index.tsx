import React from 'react';
import {
  makeStyles,
} from '@material-ui/core';
import { Formik, Form } from 'formik';
import * as yup from 'yup';
import { UPDATE } from 'queries';
import { useMutation } from '@apollo/client';
import { useViewer } from 'hooks/auth';

import InputFied from '../../InputField';
import SettingWrapper from '../../SettingWrapper';
import UpdateButton from '../../UpdateButton';
import { initialValues, UpdateUserParams } from '../utils/UpdateUtils';

const useStyles = makeStyles((theme) => ({
  containerInput: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'space-between',
    alignItems: 'flex-start',
    minHeight: '100px',
    height: '100px',
    maxHeight: '100px',
    [theme.breakpoints.down('xs')]: {
      flex: 1,
      width: '100%',
      maxHeight: '300px',
      marginBottom: theme.spacing(1),
    },
  },
  styleForm: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    flex: 1,
    width: '100%',
    [theme.breakpoints.down('xs')]: {
      flexDirection: 'column',
    },
  },
}));

const testSchema = yup.object().shape({
});

function Biography(): JSX.Element {
  const classes = useStyles();
  const viewer = useViewer();
  const [update] = useMutation(UPDATE);

  if (!viewer) {
    return null;
  }

  return (
    <SettingWrapper
      title="Biography :"
    >
      <Formik
        validationSchema={testSchema}
        initialValues={initialValues}
        onSubmit={(values: UpdateUserParams): void => {
          update({
            variables: {
              id: viewer.id,
              description: values.description,
            },
          });
        }}
      >
        <div className={classes.containerInput}>
          <Form className={classes.styleForm}>
            <InputFied
              name="description"
              label="Biography :"
              value="Software developer"
              rows={4}
              multiline
            />
            <UpdateButton description="Update your description" onClick={(): void => (console.log('save'))} />
          </Form>
        </div>
      </Formik>
    </SettingWrapper>
  );
}

export default Biography;
