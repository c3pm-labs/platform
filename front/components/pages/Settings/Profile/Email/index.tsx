import React from 'react';
import {
  makeStyles,
} from '@material-ui/core';
import { Formik, Form } from 'formik';
import * as yup from 'yup';
import { UPDATE } from 'queries';
import { useMutation } from '@apollo/client';
import { useViewer } from 'hooks/auth';

import InputFied from '../../InputField';
import SettingWrapper from '../../SettingWrapper';
import UpdateButton from '../../UpdateButton';
import { initialValues, UpdateUserParams } from '../utils/UpdateUtils';

const useStyles = makeStyles((theme) => ({
  containerInput: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'space-between',
    alignItems: 'flex-start',
    [theme.breakpoints.down('xs')]: {
      flex: 1,
      width: '100%',
      marginBottom: theme.spacing(3),
      marginTop: theme.spacing(3),
    },
  },
  styleForm: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    flex: 1,
    width: '100%',
    [theme.breakpoints.down('xs')]: {
      flexDirection: 'column',
    },
  },
}));

const testSchema = yup.object().shape({
  email: yup.string().email('Email is invalid').required('Email is required'),
});

function Email(): JSX.Element {
  const classes = useStyles();
  const viewer = useViewer();
  const [update] = useMutation(UPDATE);

  if (!viewer) {
    return null;
  }

  return (
    <SettingWrapper
      title="Email :"
    >
      <Formik
        validationSchema={testSchema}
        initialValues={initialValues}
        onSubmit={(values: UpdateUserParams): void => {
          update({
            variables: {
              id: viewer.id,
              email: values.email,
            },
          });
        }}
      >
        <div className={classes.containerInput}>
          <Form className={classes.styleForm}>
            <InputFied name="email" label="Email :" value="alice.dupont@gmail.com" />
            <UpdateButton description="Update your email" onClick={(): void => (console.log('save'))} />
          </Form>
        </div>
      </Formik>
    </SettingWrapper>
  );
}

export default Email;
