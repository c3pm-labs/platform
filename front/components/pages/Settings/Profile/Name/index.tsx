import React from 'react';
import {
  makeStyles,
} from '@material-ui/core';
import { Formik, Form } from 'formik';
import * as yup from 'yup';
import { UPDATE } from 'queries';
import { useMutation } from '@apollo/client';
import { useViewer } from 'hooks/auth';

import InputFied from '../../InputField';
import SettingWrapper from '../../SettingWrapper';
import UpdateButton from '../../UpdateButton';
import { initialValues, UpdateUserParams } from '../utils/UpdateUtils';

const useStyles = makeStyles((theme) => ({
  containerInput: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'space-between',
    alignItems: 'flex-start',
    minHeight: '100px',
    height: '100px',
    maxHeight: '100px',
    [theme.breakpoints.down('xs')]: {
      flex: 1,
      width: '100%',
      maxHeight: '300px',
    },
  },
  styleForm: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    flex: 1,
    width: '100%',
    [theme.breakpoints.down('xs')]: {
      flexDirection: 'column',
    },
  },
  textButton: {
    color: 'white',
    fontWeight: 500,
    textTransform: 'capitalize',
  },
  customButton: {
    maxHeight: '36px',
  },
}));

const testSchema = yup.object().shape({
});

function Name(): JSX.Element {
  const classes = useStyles();
  const viewer = useViewer();
  const [update] = useMutation(UPDATE);

  if (!viewer) {
    return null;
  }

  return (
    <SettingWrapper
      title="Name :"
    >
      <Formik
        validationSchema={testSchema}
        initialValues={initialValues}
        onSubmit={(values: UpdateUserParams): void => {
          update({
            variables: {
              id: viewer.id,
              username: values.username,
            },
          });
        }}
      >
        <div className={classes.containerInput}>
          <Form className={classes.styleForm}>
            <InputFied
              name="username"
              label="Username :"
              value="Alice"
              multiline
            />
            <UpdateButton description="Update your name" onClick={(): void => (console.log('save'))} />
          </Form>
        </div>
      </Formik>
    </SettingWrapper>
  );
}

export default Name;
