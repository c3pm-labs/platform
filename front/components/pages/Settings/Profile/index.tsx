import React, { useState } from 'react';
import { makeStyles, Snackbar } from '@material-ui/core';
import { Form, Formik } from 'formik';
import { useMutation } from '@apollo/client';
import * as yup from 'yup';
import { useViewer } from 'hooks/auth';
import { Alert } from '@material-ui/lab';

import { UPDATE } from '../../../../queries';
import TextInput from '../../../TextInput';
import Button from '../../../Button';

const useStyles = makeStyles((theme) => ({
  container: {
    display: 'flex',
    flex: 0.61,
    paddingLeft: theme.spacing(3),
    paddingRight: theme.spacing(3),
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: theme.spacing(3),
  },
  form: {
    display: 'flex',
    flexDirection: 'column',
    width: '100%',
    marginTop: theme.spacing(4),
  },
  line: {
    display: 'flex',
    flexDirection: 'row',
    width: '100%',
    paddingBottom: theme.spacing(4),
    alignItems: 'flex-start',
    justifyContent: 'center',
    borderBottom: `1px solid ${theme.palette.grey[400]}`,
  },
  title: {
    color: theme.palette.primary.main,
    fontSize: '1.8em',
    fontWeight: 500,
    marginBottom: theme.spacing(4),
    marginTop: theme.spacing(2),
  },
  button: {
    marginLeft: 10,
    width: 100,
    height: 40,
  },
  column: {
    display: 'flex',
    flexDirection: 'column',
    width: '100%',
    paddingBottom: theme.spacing(4),
    alignItems: 'flex-start',
    justifyContent: 'center',
    '& button': {
      marginLeft: 0,
      marginTop: 10,
    },
  },
}));

type SnackBarState = {
  type: 'success' | 'error';
  message?: string;
};

function Profile(): JSX.Element {
  const classes = useStyles();
  const viewer = useViewer();
  const [isOpen, setIsOpen] = useState<boolean>(false);
  const [snackbar, setSnackbar] = useState<SnackBarState>({ type: 'success' });
  const [update] = useMutation(UPDATE, {
    notifyOnNetworkStatusChange: true,
    onError: (error) => {
      console.log('TOTO!');
      setSnackbar({ type: 'error', message: error.message });
      setIsOpen(true);
    },
    onCompleted: () => {
      console.log('SALUT!');
      setSnackbar({ type: 'success', message: 'Infos successfully updated!' });
      setIsOpen(true);
    },
  });

  const updateSchema = yup.object().shape({
    username: yup.string().min(1).required(),
    email: yup.string().email().required(),
    description: yup.string(),
  });

  if (!viewer) return <div>Loading...</div>;

  const initialValues = {
    username: viewer?.username,
    email: viewer?.email,
    description: viewer?.description,
  };

  return (
    <div className={classes.container}>
      <Formik
        validationSchema={updateSchema}
        initialValues={initialValues}
        onSubmit={async (values): Promise<void> => {
          await update({
            variables: {
              id: viewer.id,
              ...values,
            },
          });
        }}
      >
        <Form className={classes.form}>
          <span className={classes.title}>Username :</span>
          <div className={classes.line}>
            <TextInput
              type="text"
              name="username"
              placeholder="username"
              fullWidth
            />
            <Button type="submit" variant="contained" color="primary" className={classes.button}>
              Save
            </Button>
          </div>
          <span className={classes.title}>Email :</span>
          <div className={classes.line}>
            <TextInput
              type="email"
              name="email"
              placeholder="email"
              fullWidth
            />
            <Button type="submit" variant="contained" color="primary" className={classes.button}>
              Save
            </Button>
          </div>
          <span className={classes.title}>Description :</span>
          <div className={classes.column}>
            <TextInput
              type="text"
              name="description"
              placeholder="description"
              fullWidth
              multiline
              disableHelperText
              rows={6}
            />
            <Button type="submit" variant="contained" color="primary" className={classes.button}>
              Save
            </Button>
          </div>
        </Form>
      </Formik>
      <Snackbar open={isOpen} autoHideDuration={6000} onClose={() => setIsOpen(false)} anchorOrigin={{ vertical: 'top', horizontal: 'right' }}>
        <Alert onClose={() => setIsOpen(false)} severity={snackbar.type}>
          {snackbar.message}
        </Alert>
      </Snackbar>
    </div>
  );
}

export default Profile;
