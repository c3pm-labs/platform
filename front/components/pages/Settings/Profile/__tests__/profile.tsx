import { MockedProvider } from '@apollo/client/testing';
import { render } from '@testing-library/react';

import Profile from '../index';

// eslint-disable-next-line
const useViewer = jest.spyOn(require('hooks/auth'), 'useViewer');

test('Profile', () => {
  const mocks = [];

  useViewer.mockImplementation(() => ({
    username: 'toto',
    email: 'tata',
  }));

  const { container } = render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <Profile />
    </MockedProvider>,
  );

  expect(container).toMatchSnapshot();
});
