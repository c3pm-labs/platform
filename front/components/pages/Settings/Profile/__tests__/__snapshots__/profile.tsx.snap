// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile 1`] = `
<div>
  <div
    class="makeStyles-container-1"
  >
    <form
      action="#"
      class="makeStyles-form-2"
    >
      <span
        class="makeStyles-title-4"
      >
        Username :
      </span>
      <div
        class="makeStyles-line-3"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <input
              aria-describedby="username-helper-text"
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              id="username"
              name="username"
              placeholder="username"
              type="text"
              value="toto"
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-9"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            class="MuiFormHelperText-root makeStyles-root-7 MuiFormHelperText-contained MuiFormHelperText-filled MuiFormHelperText-marginDense"
            id="username-helper-text"
          >
            <span>
              ​
            </span>
          </p>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-default-12 makeStyles-primary-13 makeStyles-primary-17 makeStyles-button-5 MuiButton-disableElevation"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Save
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <span
        class="makeStyles-title-4"
      >
        Email :
      </span>
      <div
        class="makeStyles-line-3"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <input
              aria-describedby="email-helper-text"
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              id="email"
              name="email"
              placeholder="email"
              type="email"
              value="tata"
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-9"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            class="MuiFormHelperText-root makeStyles-root-7 MuiFormHelperText-contained MuiFormHelperText-filled MuiFormHelperText-marginDense"
            id="email-helper-text"
          >
            <span>
              ​
            </span>
          </p>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-default-12 makeStyles-primary-13 makeStyles-primary-21 makeStyles-button-5 MuiButton-disableElevation"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Save
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <span
        class="makeStyles-title-4"
      >
        Description :
      </span>
      <div
        class="makeStyles-column-6"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <textarea
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              id="description"
              name="description"
              placeholder="description"
              rows="6"
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-9"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
          
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-default-12 makeStyles-primary-13 makeStyles-primary-25 makeStyles-button-5 MuiButton-disableElevation"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Save
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </form>
  </div>
</div>
`;
